PYME/Acquire/Hardware/Simulator/fluor.py

```diff
@@ -80,17 +79,17 @@ def __init__(self, x, y, z, transitionProbablilities, excitationCrossections, th
         self.state = initialState
         self.activeState = activeState
         self.thetas = thetas
-        self.transitionProbabilities = transitionProbablilities * concatenate(([1], abs(cos(thetas))),0)
-        self.excitationCrossections = excitationCrossections *abs(cos(thetas))
+        self.transitionProbabilities = transitionProbablilities * np.concatenate(([1], abs(np.cos(thetas))),0)
+        self.excitationCrossections = excitationCrossections *abs(np.cos(thetas))
 
     def illuminate(self, laserPowers, expTime):
-        dose = concatenate(([1],laserPowers),0)*expTime
+        dose = np.concatenate(([1],laserPowers),0)*expTime
         #grab transition matrix
         transVec = (self.transitionProbabilities[self.state,:,:]*dose).sum(1)
         transVec[self.state]= 1 - transVec.sum()
         transCs = transVec.cumsum()
         
-        r = rand()
+        r = np.random.RandomState().rand()  # replace with np.random.default_rng() on later numpy versions
```

Comment:
`np.random.rand()` should work fine (alternatively, `np.random.uniform()`)
```suggestion
        r = np.random.rand() 
```

Reply:
Correct, but in the numpy docs, there is a note that `random.rand` is a convenience wrapper for people coming from matlab, and `random_sample` takes a tuple size more consistent with the rest of numpy. On the docs for `random_sample` there's a see also for `Generator.random` which "should be used for new code", and would be `np.random.default_rng().random()`. However, that is not available until numpy 1.17, hence the `random.RandomState`. 

All that's to say - I'm fine using `random.rand`. More info [here](https://numpy.org/doc/stable/reference/random/new-or-different.html#new-or-different) when we get to it.

Reply:
note also, the [scipy deprecation notes](https://docs.scipy.org/doc/scipy/reference/release.1.4.0.html#scipy-deprecations) specifically say to use `default_rng`